api/api.go

```diff
@@ -9,22 +9,25 @@ import (
 	"github.com/mesg-foundation/core/event"
 	"github.com/mesg-foundation/core/execution"
 	"github.com/mesg-foundation/core/service"
+	"github.com/mesg-foundation/core/service/manager"
 	uuid "github.com/satori/go.uuid"
 )
 
 // API exposes all functionalities of MESG core.
 type API struct {
 	ps *pubsub.PubSub
 
+	m         manager.Manager
+	container container.Container
```

Comment:
I think container should not be needed anymore in the api because the logic is delegated to the manager

Reply:
it's needed for deploy

Reply:
and don't you think some part of the deploy make sense to have in the manager with a `prepare/init/build/create` function and have the deploy like that 
```
deploy() {
  service = import.From(...)
  metaInfo = manager.Prepare(service)
  db.Save(metaInfo)
  db.Save(service)
}
```

Reply:
you mean actually create the docker image inside an implementation of `manager.Manager`? what do you mean about `prepare/init/build/create`? i understand the `build` part, what do you mean by the other ones?

for the build, it cannot be inside of the implementations of `manager.Manager` type. because it's not specific the orchestration tool that used. it's a generic thing for all. but sure, it can be moved inside to the `manager` pkg as a standalone func but not a part of `manager.Manager` interface.

Reply:
What do you mean it's not specific to the orchestration tool? There are different way to create image for different orchestration/containerization tool.

`dockermanager` should implement its own build function.
But if we have multiple `manager` that actually use the same build function, let's say a `dockermanager` and a `k8manager`, then they should use a common function from another package, something like `containerd.build`.
Let's not force this 'factorization' at the higher level. It should be at the low level.

I think here, the goal is to reduce the dependency of the `API` package and delegate more to other package.

Reply:
Let's continue this conversation outside this PR: https://github.com/mesg-foundation/core/issues/956